dll/win32/shell32/wine/pidl.c

```diff
@@ -1722,6 +1722,80 @@ LPITEMIDLIST _ILCreateGuidFromStrA(LPCSTR szGUID)
 }
 #endif
 
+// Create GUID from shortcut(desktop, Personal...)
+// returns TRUE if succefull and FALSE otherwise
+BOOL _ILCreateGuidFromShortcutW(LPCWSTR szShortcut, GUID *fidOut)
+{
+	if (!StrCmpIW(szShortcut, L"ADMINISTRATIVE TOOLS"))
```

Comment:
I wonder whether (especially if more to be added in the future) you could use a kind of table "lookup" that would avoid code repetition, something like this:
```c
static const struct {
    LPCWSTR pszName;
    GUID guid;
} ShellShortcuts[] =
{
    {L"CONTROLPANELFOLDER", CLSID_ControlPanel},
    {L"FONTS", CLSID_FontsFolderShortcut},
/* et caetera, you see the idea... */
};

    UINT i;
    for (i = 0; i < _countof(ShellShortcuts); ++i)
    {
        if (StrCmpIW(szShortcut, ShellShortcuts[i].pszName) == 0)
        {
            *fidOut = ShellShortcuts[i].guid;
            return TRUE;
            return TRUE;
        }
    }

    return FALSE;
```